### PR TITLE
virt-net: handle more than one vnic backing device

### PR DESCRIPTION
**Pull request by cforno12**
Currently, the get_failover_priority only checks if the first vNIC device has
the slot number. There is the case where the lshwres command retrieves multiple
vNIC devices. The output is formatted where it creates an array of all the
vNIC devices and checks if the slot number matches each one.

Signed-off-by: Cristobal Forno <cforno12@linux.ibm.com>

### CODE REVIEW

**Comment by smruti77** [comment]
@cforno12  if it will create more than one vnic backing device, need to handle the same in backing device remove to delete multiple vnic backing device.

**Comment by cforno12** [comment]
@smruti77 I think I did a bad job describing this fix. 
Let me rephrase, if there is an LPAR with more than one vNIC interface, the lshwres command will output something like this:
```
3,sriov/ltczep10-vios1/100/2/0/27008002/2.0/2.0/50/100.0/100.0
8,"sriov/ltczep10-vios1/100/2/3/27008003/24.0/24.0/1/100.0/100.0,sriov/ltczep10-vios1/100/2/2/27008005/24.0/24.0/100/100.0/100.0"
```
With the current code, it will see that the first line is not equal to our targeted vNIC interface (which has slot 8) and will return with `None`. My patch will parse the second line of the output as we are focusing on the interface that lies on slot 8.

From my understanding, we do not need to handle removing multiple vNIC interfaces as we are only adding and removing the one, in my case the one in slot 8. Please let me know if I am misunderstanding what you are getting at. Thanks for reviewing.

**Comment by narasimhan-v** [comment]
@cforno12 How about using slot number also in `--filter` section ?

```
hscroot@ltchmcv4:~> lshwres -r virtualio -m ltcfleet2 --rsubtype vnic --level lpar --filter lpar_names=ltcfleet2-lp12-vaishnavi -F slot_num,backing_devices
3,sriov/ltcfleet2-vios1/100/6/0/27018019/2.0/2.0/50/100.0/100.0
hscroot@ltchmcv4:~> lshwres -r virtualio -m ltcfleet2 --rsubtype vnic --level lpar --filter slots=3,lpar_names=ltcfleet2-lp12-vaishnavi -F slot_num,backing_devices
3,sriov/ltcfleet2-vios1/100/6/0/27018019/2.0/2.0/50/100.0/100.0
hscroot@ltchmcv4:~> lshwres -r virtualio -m ltcfleet2 --rsubtype vnic --level lpar --filter slots=4,lpar_names=ltcfleet2-lp12-vaishnavi -F slot_num,backing_devices
No results were found.
hscroot@ltchmcv4:~> 
```

**Comment by abdhaleegit** [comment]
@cforno12 Yes change the command to get specific slot using slots filter

**Comment by smruti77** [comment]
> @smruti77 I think I did a bad job describing this fix.
> Let me rephrase, if there is an LPAR with more than one vNIC interface, the lshwres command will output something like this:
> 
> ```
> 3,sriov/ltczep10-vios1/100/2/0/27008002/2.0/2.0/50/100.0/100.0
> 8,"sriov/ltczep10-vios1/100/2/3/27008003/24.0/24.0/1/100.0/100.0,sriov/ltczep10-vios1/100/2/2/27008005/24.0/24.0/100/100.0/100.0"
> ```
> 
> With the current code, it will see that the first line is not equal to our targeted vNIC interface (which has slot 8) and will return with `None`. My patch will parse the second line of the output as we are focusing on the interface that lies on slot 8.
> 
> From my understanding, we do not need to handle removing multiple vNIC interfaces as we are only adding and removing the one, in my case the one in slot 8. Please let me know if I am misunderstanding what you are getting at. Thanks for reviewing.

@cforno12 ok got it, i thought it will create more than one backing device based on headline.

**Comment by cforno12** [comment]
LOGS:
```
2020-11-24 17:23:08,982 process          L0604 INFO | Running '/usr/bin/ssh -o 'StrictHostKeyChecking=no' -o 'UpdateHostKeys=no' -o 'ControlPath=~/.ssh/avocado-master-%r@%h:%p' -l hscroot -q 9.3.90.230 'lshwres -r virtualio -m ltczep10 --rsubtype vnic --level lpar           \
            --filter slots=6,lpar_names=ltczep10-lp3 -F slot_num,backing_devices''
2020-11-24 17:23:09,007 process          L0416 DEBUG| [stdout] 6,"sriov/ltczep10-vios1/100/2/2/27008020/24.0/24.0/1/100.0/100.0,sriov/ltczep10-vios1/100/2/3/27008021/24.0/24.0/100/100.0/100.0"
2020-11-24 17:23:09,008 process          L0686 INFO | Command '/usr/bin/ssh -o 'StrictHostKeyChecking=no' -o 'UpdateHostKeys=no' -o 'ControlPath=~/.ssh/avocado-master-%r@%h:%p' -l hscroot -q 9.3.90.230 'lshwres -r virtualio -m ltczep10 --rsubtype vnic --level lpar           \
            --filter slots=6,lpar_names=ltczep10-lp3 -F slot_num,backing_devices'' finished with 0 after 0.02393937110900879s
```

**Review by smruti77 — APPROVED** [review]

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Review by narasimhan-v — APPROVED** [review]
